kura/examples/org.eclipse.kura.raspberrypi.sensehat.example/src/main/java/org/eclipse/kura/raspberrypi/sensehat/example/SenseHatExample.java

```diff
@@ -0,0 +1,443 @@
+		if (m_properties.get(HUM_ENABLE) != null)
+			m_humEnable = (Boolean) m_properties.get(HUM_ENABLE);
+		if (m_properties.get(LCD_ENABLE) != null)
+			m_lcdEnable = (Boolean) m_properties.get(LCD_ENABLE);
+		if (m_properties.get(STICK_ENABLE) != null)
+			m_stickEnable = (Boolean) m_properties.get(STICK_ENABLE);
+		if (m_properties.get(SCREEN_MESSAGE) != null)
+			m_screenMessage = (String) m_properties.get(SCREEN_MESSAGE);		
+		if (m_properties.get(SCREEN_ROTATION) != null)
+			m_screenRotation = (Integer) m_properties.get(SCREEN_ROTATION);
+		if (m_properties.get(SCREEN_TEXT_COLOR) != null) {
+			if (((String) m_properties.get(SCREEN_TEXT_COLOR)).contains("RED")) 
+				m_screenTextColor = Colors.RED;
+			else if (((String) m_properties.get(SCREEN_TEXT_COLOR)).contains("ORANGE")) 
+				m_screenTextColor = Colors.ORANGE;
+			else if (((String) m_properties.get(SCREEN_TEXT_COLOR)).contains("YELLOW")) 
+				m_screenTextColor = Colors.YELLOW;
+			else if (((String) m_properties.get(SCREEN_TEXT_COLOR)).contains("GREEN")) 
+				m_screenTextColor = Colors.GREEN;
+			else if (((String) m_properties.get(SCREEN_TEXT_COLOR)).contains("BLUE")) 
+				m_screenTextColor = Colors.BLUE;
+			else if (((String) m_properties.get(SCREEN_TEXT_COLOR)).contains("PURPLE")) 
+				m_screenTextColor = Colors.PURPLE;
+			else if (((String) m_properties.get(SCREEN_TEXT_COLOR)).contains("VIOLET")) 
+				m_screenTextColor = Colors.VIOLET;
+			else if (((String) m_properties.get(SCREEN_TEXT_COLOR)).contains("WHITE")) 
+				m_screenTextColor = Colors.WHITE;
+			else if (((String) m_properties.get(SCREEN_TEXT_COLOR)).contains("BLACK")) 
+				m_screenTextColor = Colors.BLACK;
+		}
+
+		if (startUpdateThread != null) {
+			startUpdateThread.cancel(true);
+			startUpdateThread = null;
+		}
+		startUpdateThread = m_executor.schedule(new Runnable() {
+			public void run() {
+				update();
+			}
+		}, 0, TimeUnit.MILLISECONDS);
+		
+		s_logger.info("Activating Sense Hat Application... Done.");
+	}
+
+
+	protected void deactivate(ComponentContext componentContext) 
+	{
+		s_logger.info("Deactivating Sense Hat Application...");
+
+		LPS25H.closeDevice();
+		HTS221.closeDevice();
+		LSM9DS1.closeDevice();
+
+		if (m_Joystickhandle != null)
+			m_Joystickhandle.cancel(true);
+		if (m_Joystickworker != null)
+			m_Joystickworker.shutdown();
+		if (SenseHatJoystick != null)
+			Joystick.closeJoystick();
+
+
+		if (frameBuffer != null) {
+			frameBuffer.clearFrameBuffer();
+			FrameBuffer.closeFrameBuffer();
+			frameBuffer = null;
+		}
+
+		if (startUpdateThread != null) {
+			startUpdateThread.cancel(true);
+			startUpdateThread = null;
+		}
+		m_executor = null;
+		
+		s_logger.info("Deactivating Sense Hat Application... Done.");
+	}	
+
+
+	public void updated(Map<String,Object> properties)
+	{
+		s_logger.info("Updated Sense Hat Application...");
+
+		// store the properties received
+		m_properties = properties;
+		if (m_properties.get(IMU_ACC_ENABLE) != null) 
```

Comment:
Uhm, code duplication with activate method? Maybe we can extract to a method or call updated from activated.